### PR TITLE
Fix anon chat, tab completion and other small stuff

### LINKED ISSUE #2402: Anon chat isn't working properly
I usually use anon chat when on Twitch, and it usually lets me go into a stream invisible, but the last few days it hasn't been working.  It shows me in the chat list name and I have to click on anon chat to get it to work again.

### PR DESCRIPTION
**Pull request by EhsanKia**
Closes #2402, #2397, #2390, #2333

### CODE REVIEW

File: src/modules/better_viewer_list/index.js [hunk 1]
```diff
@@ -255,6 +254,10 @@ class BetterViewerListModule {
         // Load viewers
         this.loadViewerList();
     }
+
+    capitalize(str) {
```
Comment (night) [inline src/modules/better_viewer_list/index.js]: this doesn't really need to be on the class.. i'd drop it as a named function at the top

Comment (EhsanKia) [inline src/modules/better_viewer_list/index.js]: done.

File: src/modules/better_viewer_list/index.js [hunk 2]
```diff
@@ -89,6 +89,7 @@ class BetterViewerListModule {
     extractViewers(data) {
         const typeDisplays = ['STAFF', 'ADMINS', 'GLOBAL MODERATORS', 'MODERATORS', 'VIEWERS'];
         const userTypes = ['staff', 'admins', 'global_mods', 'moderators', 'viewers'];
+        const capitalize = str => str.charAt(0).toUpperCase() + str.slice(1);
```
Comment (night) [inline src/modules/better_viewer_list/index.js]: but why? :notlikethis: i meant before the class is even defined

Comment (EhsanKia) [inline src/modules/better_viewer_list/index.js]: oh oops, :haHAA:

File: src/modules/chat/index.js [hunk 3]
```diff
@@ -12,7 +12,7 @@ const channelEmotesTip = require('../channel_emotes_tip');
 const legacySubscribers = require('../legacy_subscribers');
 
 const EMOTE_STRIP_SYMBOLS_REGEX = /(^[~!@#$%\^&\*\(\)]+|[~!@#$%\^&\*\(\)]+$)/g;
-const MENTION_REGEX = /@\b(.*)\b/;
+const MENTION_REGEX = /^@([a-zA-Z\d_]+)$/;
```
Comment (night) [inline src/modules/chat/index.js]: i think this breaks for i18n names.. should we just match until end or symbol?

Comment (EhsanKia) [inline src/modules/chat/index.js]: Err, are we doing @ mention for displaynames or just usernames? How does that even work? I mean, clicking on it doesn't work right? Do you just care about the slight bolding?

**Pull request merged**